synedrion/src/cggmp21/sigma/dec.rs

```diff
@@ -58,15 +62,18 @@ impl<P: SchemeParams> DecProof<P> {
 
         let hat_cap_n = &setup.modulus_bounded(); // $\hat{N}$
 
-        let alpha = Signed::random_bounded_bits(rng, P::L_BOUND + P::EPS_BOUND);
-        let mu = Signed::random_bounded_bits_scaled(rng, P::L_BOUND, hat_cap_n);
-        let nu = Signed::random_bounded_bits_scaled(rng, P::L_BOUND + P::EPS_BOUND, hat_cap_n);
+        let alpha = Secret::init_with(|| Signed::random_bounded_bits(rng, P::L_BOUND + P::EPS_BOUND));
+        let mu = Secret::init_with(|| Signed::random_bounded_bits_scaled(rng, P::L_BOUND, hat_cap_n));
+        let nu = Secret::init_with(|| Signed::random_bounded_bits_scaled(rng, P::L_BOUND + P::EPS_BOUND, hat_cap_n));
         let r = Randomizer::random(rng, pk0);
 
         let cap_s = setup.commit(y, &mu).to_wire();
         let cap_t = setup.commit(&alpha, &nu).to_wire();
-        let cap_a = Ciphertext::new_with_randomizer_signed(pk0, &alpha, &r.to_wire()).to_wire();
-        let gamma = P::scalar_from_signed(&alpha);
+        let cap_a = Ciphertext::new_with_randomizer_signed(pk0, &alpha, &r).to_wire();
+
+        // `alpha` is secret, but `gamma` only uncovers $\ell$ bits of `alpha`'s full $\ell + \eps$ bits,
```

Comment:
So 256 bits of 3*256 bits? (`L_BOUND` = 256 and `EPS_BOUND` = 2*`L_BOUND`)?

Reply:
Yep. In any case, it's going in the proof which is supposed to be public, so I guess it shouldn't be a problem for it to be exposed.